File: core/src/main/java/com/core/BaseActivity.java
package com.core;

import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import android.content.Context;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import android.os.Bundle;
import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import android.view.WindowManager;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;

public abstract class BaseActivity<VM extends ViewModel, VDB extends ViewDataBinding> extends DaggerAppCompatActivity implements BaseView, LifecycleOwner {

    @Inject
    protected ViewModelProvider.Factory viewModelFactory;

    private AlertDialog dialog;

    protected VM viewModel;

    private VDB binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, getLayoutResId());
        binding.setLifecycleOwner(this);

        viewModel = ViewModelProviders
                .of(this, viewModelFactory)
                .get(getViewModelClass());

        onInitialized(savedInstanceState, viewModel, binding);
    }

    protected abstract Class<VM> getViewModelClass();

    @LayoutRes
    protected abstract int getLayoutResId();

    protected abstract void onInitialized(Bundle instance, VM viewModel, VDB binding);

    @Override
    public void showLoadingBar() {
        if (dialog == null)
            dialog = new AlertDialog.Builder(context(), R.style.DialogStyle)
                    .setView(R.layout.view_progress)
                    .setCancelable(false)
                    .create();

        if(dialog.getWindow() != null)
            dialog.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);

        if (dialog.isShowing())
            return;

        dialog.show();
    }

    @Override
    public void hideLoadingBar() {
        dialog.dismiss();
    }

    @Override
    public Context context() {
        return this;
    }

    public VDB getBinding() {
        return binding;
    }

    public VM getViewModel() {
        return viewModel;
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/base/activity/BaseActivity.kt
package com.volkanhotur.basemvvm.base.activity

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.databinding.ViewDataBinding
import androidx.lifecycle.ViewModel
import com.volkanhotur.basemvvm.base.exception.UnCaughtExceptionHandler
import com.volkanhotur.basemvvm.utils.helper.LocaleHelper
import com.volkanhotur.basemvvm.utils.impl.DefaultClickHandler
import com.core.abstractbase.AbstractActivity

/**
 * @author volkanhotur
 */

abstract class BaseActivity<VM : ViewModel, VDB : ViewDataBinding> : AbstractActivity<VM, VDB>(), DefaultClickHandler {

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(LocaleHelper.onAttach(base))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Thread.setDefaultUncaughtExceptionHandler(UnCaughtExceptionHandler(this))
        super.onCreate(savedInstanceState)
    }

    override fun navigateView() {}

    override fun expireSession(errorTitle : String?, errorMessage: String?) {
    }

    override val isLoadingBarEnabled: Boolean
        get() = true

    override fun onClick(view: View?) {}
}